services/exam-manager/app/exam.py

```diff
@@ -75,44 +75,46 @@ async def create_exam_template(payload: BaseExam) -> ExamOut:
     HTTPException
         404: Creation unsuccessful
     """
-    if not payload.is_template:
-        raise HTTPException(status_code=404, detail="Exam is not a template.")
     if not (exam := await dal.add_exam_data(payload=payload)):
         raise HTTPException(status_code=404, detail="Could not create exam")
     return await get_exam_out_model(data=exam)
 
 
 @router.post("/", response_model=ExamOut, status_code=201, tags=["exams"])
-async def create_exam_from_template(patient_id: int, template_id: UUID) -> ExamOut:
-    """Create a new exam instance from template.
+async def create_exam_from_template(patient_id: int, template_id: UUID, new_exam_is_template: bool) -> ExamOut:
```

Comment:
It allows to 1. create a template from a template essentially copying it and 2. to create an instance from a template. I think both cases make sense.

Reply:
As templates and instances share the same table, template_id could also reference to an instance. I'm not sure, if we want to allow this, for now I would add a check, if the template_id is really a template ID.
And I would suggest to rename the function.

Reply:
I could add a check to make sure only templates are copied. However I don't think there is a problem with copying instances, as long as we reset fields like the status. I'll have a look at that.
But the feature to create a template from another template is quite handy I find and I would like to keep it. For example you might want to copy a workflow template to several exam templates etc. I also thought about changing the relations such that a workflow can be part of multiple exams, but that gets confusing too if you are not very careful with designing the frontend for it.

Reply:
Not sure if I understood your point correctly, but I think it is not necessary to have multiple copies of a template and thus we don't need to copy them. E.g. a single task template can be referenced in multiple workflow templates. Only at instance creation multiple instance-copies (created from the template) would be created.
Of course we can allow to copy instances, but then we should take special care of it, as you suggested.
I think we discussed working with references (instead of copies) during the last hackathon and agreed that it is too difficult/unsafe.

Reply:
Changed create exam from template such that it always sets the status of the new exam to 'NEW'.
I created a new issue for the other points, because there are more checks missing than we discussed here ([Issue 133](https://github.com/brain-link/scanhub/issues/133))